### ISSUE TITLE
PyTorch CATENets inconsistent with PyTorch APIs

### ISSUE THREAD

**Issue by kamilest** [AliciaCurth/CATENets#19]
The PyTorch `BaseCATEEstimator`, `PropensityNet`, `RepresentationNet`, `BasicNet` (and possibly other models based on these) have a `train` method that overrides the [`torch.nn.Module.train()`](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.train), intended to set the mode to training (affecting the behaviour of Dropout, BatchNorm, etc.)

At the same time, in the actual training procedures of these models the mode is never explicitly set to training.

I would propose getting rid of the `train` methods in the `BaseCATEEstimator` interface and elsewhere, and fixing the behaviour in the corresponding `fit` methods (but currently don't have much time to look into the dependencies this would cause in experiments and so on).
